### PR TITLE
Support symbolic link and file permission mode

### PR DESCRIPTION
**Pull request by popcornylu**
- Symbolic link is now supported
- The file permission mode is currently supported.

### CODE REVIEW

File: internal/core/manager.go [hunk 1]
```diff
@@ -391,6 +391,11 @@ func (mngr *ArtifactManager) Push(options PushOptions) error {
 	mapUploadBlob := map[string]DiffRecord{}
 	for _, record := range result.Records {
 		if record.Type == DiffTypeAdd || record.Type == DiffTypeChange {
+			if record.Hash == "" {
```
Comment (wcchang1115) [inline internal/core/manager.go:394]: maybe we can encapsulate symlink check to a function in our later development (e.g. record.isSymlink())

Comment (popcornylu) [reply internal/core/manager.go:394]: The record is DiffRecord. I think the `isSymbolic()` is bit confusing in different diff types.

File: internal/core/manager.go [hunk 2]
```diff
@@ -685,19 +696,58 @@ func (mngr *ArtifactManager) Pull(options PullOptions) error {
...
+
 		switch record.Type {
+		case DiffTypeAdd:
+			if record.Link != "" {
+				err := symlinkFile(record.Link, absPath)
+				if err != nil {
+					return err
+				}
+			} else {
+				err := chmod(absPath, mode)
```
Comment (wcchang1115) [inline internal/core/manager.go:712]: I think it will not be reached, but it won't cause issues.

File: internal/core/manager.go (hunk 1 above)
Comment (wcchang1115) [reply internal/core/manager.go:394]: yeah, agree with you -> there is a DiffRecord context behind it.

**Review by wcchang1115 — APPROVED** [review]

**Pull request merged**